### DISCUSSION TITLE
Unable to communicate Orc8r with Kubevirt VM AGW

### DISCUSSION THREAD

**Discussion by pmishra-wavelabs** [magma/magma#6119] [Q&A]
I have installed nms and orc8r containers on debian 9 machine.
And my AGW is on Kubevirt VM.
I am getting issue While Orc8r communicating with AGW.
=====Error=====
sudo journalctl -fu magma@magmad
-- Logs begin at Sun 2021-04-11 23:41:36 UTC. --
Apr 13 07:42:36 debian magmad[2085]: INFO:root:Starting magmad...
Apr 13 07:42:36 debian magmad[2085]: INFO:root:Listening on address 127.0.0.1:50052
Apr 13 07:42:36 debian magmad[2085]: INFO:root:Cannot load a proper cert, start bootstrapping
Apr 13 07:42:36 debian magmad[2085]: INFO:root:Waiting before checking for updates for the first time...
Apr 13 07:42:36 debian magmad[2085]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] Name resolution failure
Apr 13 07:42:36 debian magmad[2085]: INFO:root:Retrying bootstrap in 30 seconds
Apr 13 07:42:36 debian magmad[2085]: ERROR:root:Checkin Error! Failed to report states. [StatusCode.CANCELLED] Received http2 header with status: 503
Apr 13 07:42:36 debian magmad[2085]: INFO:root:StateReporting (Checkin) failure threshold met, remediating...
Apr 13 07:42:37 debian magmad[2085]: ERROR:root:Error making TCP connection: Unknown error -2, [Errno -2] Name or service not known
Apr 13 07:42:37 debian magmad[2085]: ERROR:root:StateReporting failure likely not due to invalid cert
Apr 13 07:43:06 debian magmad[2085]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] Name resolution failure
Apr 13 07:43:06 debian magmad[2085]: INFO:root:Retrying bootstrap in 30 seconds
Apr 13 07:43:36 debian magmad[2085]: ERROR:root:Metrics upload error! [StatusCode.CANCELLED] Received http2 header with status: 503
Apr 13 07:43:36 debian magmad[2085]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] Name resolution failure
Apr 13 07:43:36 debian magmad[2085]: INFO:root:Retrying bootstrap in 30 seconds
Apr 13 07:43:36 debian magmad[2085]: ERROR:root:Checkin Error! Failed to report states. [StatusCode.CANCELLED] Received http2 header with status: 503
Apr 13 07:43:36 debian magmad[2085]: INFO:root:StateReporting (Checkin) failure threshold met, remediating...
Apr 13 07:43:36 debian magmad[2085]: ERROR:root:Error making TCP connection: Unknown error -2, [Errno -2] Name or service not known
Apr 13 07:43:36 debian magmad[2085]: ERROR:root:StateReporting failure likely not due to invalid cert

As per my understanding:
DNS setup is fine.
On file : /etc/magma/control_proxy.yml

rootca_cert: /var/opt/magma/tmp/certs/rootCA.pem
cloud_address: controller.mydomain.com
cloud_port: 8443 // Also tried with 443,7443,9443
bootstrap_address: bootstrapper-controller.mydomain.com
bootstrap_port: 8444 // Also tried with 443,7444,9444

checkin_cli.py 
Output : 
/usr/local/lib/python3.5/dist-packages/magma/common/cert_utils.py:16: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release of cryptography. Please upgrade your Python.
  from cryptography import x509
1. -- Testing TCP connection to controller.mydomain.com:443 --
> Error: [Errno -2] Name or service not known
Suggestions:
            - Verify hostname in /etc/magma/control_proxy.yml &
              /var/opt/magma/configs/control_proxy.yml.
            - Verify correct port.
            - Check DNS (nslookup hostname).
            - Make sure you are disconnected from VPN.
            - Check nghttpx is running on cloud VM.

Python Version : 3.7.3 @parthiban337 @karthiksubraveti @GANESH-WAVELABS

**Comment by karthiksubraveti** [comment]
@pmishra-wavelabs This clearly seems to be like a DNS issue. 
Add the following entries to /etc/hosts
HOST_MACHINE_IP controller.magma.test
HOST_MACHINE_IP  bootstrapper-controller.magma.test
This is similar to https://github.com/magma/magma/discussions/5814

**Reply by pmishra-wavelabs** [reply]
Thanks For reply. 
On /etc/hosts file.
10.9.9.46	magma-test.magma.wavelabs.in
10.9.9.46	master.magma.wavelabs.in
10.9.9.46	magma.magma.wavelabs.in
10.9.9.46	magma
10.9.9.46	master
10.9.9.46	nms.magma.wavelabs.in
10.9.9.46	controller.magma.wavelabs.in
10.9.9.46	bootstrapper-controller.magma.wavelabs.in
10.9.9.46	api.magma.wavelabs.in
10.9.9.46	magma.test
10.9.9.46	magma-test.nms.magma.wavelabs.in
########
10.9.9.46	controller.magma.test
10.9.9.46	bootstrapper-controller.magma.test

**Comment by karthiksubraveti** [comment]
may be a silly question here. Did you replace your control proxy 'mydomain' -> with your actual domain ?
rootca_cert: /var/opt/magma/tmp/certs/rootCA.pem
cloud_address: controller.mydomain.com
cloud_port: 8443 // Also tried with 443,7443,9443
bootstrap_address: bootstrapper-controller.mydomain.com
bootstrap_port: 8444 // Also tried with 443,7444,9444

**Reply by pmishra-wavelabs** [reply]
# Location for certs
rootca_cert: /var/opt/magma/tmp/certs/rootCA.pem
local_port: 8443
cloud_address: controller.magma.test // tried with **controller.magma-test.wavelabs.in** also getting same error
cloud_port: 7443
bootstrap_address: bootstrapper-controller.magma.test // tried with **bootstrapper-controller.magma-test.wavelabs.in** also getting same error
bootstrap_port: 7444

sudo journalctl -fu magma@magmad
-- Logs begin at Tue 2021-04-13 03:11:42 UTC. --
Apr 14 13:36:56 debian magmad[12549]: E0414 13:36:56.439445913   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:36:57 debian magmad[12549]: E0414 13:36:57.511829572   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:00 debian magmad[12549]: E0414 13:37:00.057354956   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:06 debian magmad[12549]: E0414 13:37:06.351434264   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:09 debian magmad[12549]: E0414 13:37:09.774562587   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:18 debian magmad[12549]: E0414 13:37:18.848762395   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:21 debian magmad[12549]: E0414 13:37:21.151880720   13222 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:21 debian magmad[12549]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] Connect Failed
Apr 14 13:37:21 debian magmad[12549]: INFO:root:Retrying bootstrap in 30 seconds
Apr 14 13:37:22 debian magmad[12549]: E0414 13:37:22.143352695   12559 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:23 debian magmad[12549]: E0414 13:37:23.733983567   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:24 debian magmad[12549]: E0414 13:37:24.241380266   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:25 debian magmad[12549]: E0414 13:37:25.849586297   12559 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
Apr 14 13:37:29 debian magmad[12549]: E0414 13:37:29.840858958   12564 ssl_transport_security.cc:1229] Handshake failed with fatal error SSL_ERROR_SSL: error:10000095:SSL routines:OPENSSL_internal:ERROR_PARSING_EXTENSION.
^C
magma@debian:~$ nano /etc/magma/control_proxy.yml
magma@debian:~$ ls /etc/ssl/certs/ca-certificates.crt
/etc/ssl/certs/ca-certificates.crt
magma@debian:~$

**Comment by pmishra-wavelabs** [comment]
Hi Karthik,
Any more suggestions about the issue.?
I tried all the steps given the magma documentation & also suggestions given in Slack Channel. Can you help here to resolve this....

**Comment by karthiksubraveti** [comment]
@pmishra-wavelabs can you dump the current state of your /etc/hosts file and your control_proxy file. Also can you dump the logs on orc8r. Now we seem to be seeing different error than name resolution failure we saw earlier. It mentions that the connect is failing. Is the orc8r pod crashing ?

**Reply by pmishra-wavelabs** [reply]
Thanks Karthik for your response. Can You please help me out on the manual way for "HOST [magma/lte/gateway]$ fab -f dev_tools.py register_vm". As this is for Vagrant VM. 
Today I have recreated every thing from scratch with Kubevirt. Only thing b/w vagrant and kubevirt Fabric command **fab -f dev_tools.py register_vm**, I can't apply. Instead I use below link :
https://docs.magmacore.org/docs/next/lte/deploy_config_agw
Orc8r logs I am attaching
**Under AGW(Same in Local machine) : /etc/hosts**

127.0.1.1 debian debian
127.0.0.1 localhost
#######################
10.9.9.46    magma-test.magma.wavelabs.in
10.9.9.46    master.magma.wavelabs.in
10.9.9.46    nms.magma.wavelabs.in
10.9.9.46    controller.magma.wavelabs.in
10.9.9.46    bootstrapper-controller.magma.wavelabs.in
10.9.9.46    api.magma.wavelabs.in
10.9.9.46    magma.test
########
10.9.9.46    controller.magma.test
10.9.9.46    bootstrapper-controller.magma.test
# The following lines are desirable for IPv6 capable hosts
::1 ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts

=====================================================================
**magma@debian:~$ checkin_cli.py**
/usr/local/lib/python3.5/dist-packages/magma/common/cert_utils.py:16: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release of cryptography. Please upgrade your Python.
  from cryptography import x509
1. -- Testing TCP connection to controller.magma.wavelabs.in:7443 --
> Error: [Errno -2] Name or service not known
Suggestions:
            - Verify hostname in /etc/magma/control_proxy.yml &
              /var/opt/magma/configs/control_proxy.yml.
            - Verify correct port.
            - Check DNS (nslookup hostname).
            - Make sure you are disconnected from VPN.
            - Check nghttpx is running on cloud VM.
=====================================================================
**magma@debian:~$ sudo journalctl -fu magma@magmad**
-- Logs begin at Sat 2021-04-17 12:41:16 UTC. --
Apr 17 12:52:47 debian magmad[12496]: ERROR:root:GetOperationalStates Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:52:47 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:52:47 debian magmad[12496]: ERROR:root:Checkin Error! Failed to report states. [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:52:47 debian magmad[12496]: INFO:root:StateReporting (Checkin) failure threshold met, remediating...
Apr 17 12:52:47 debian magmad[12496]: ERROR:root:Error making TCP connection: Unknown error -2, [Errno -2] Name or service not known
Apr 17 12:52:47 debian magmad[12496]: ERROR:root:StateReporting failure likely not due to invalid cert
Apr 17 12:52:57 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:07 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:15 debian magmad[12496]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] DNS resolution failed for service: bootstrapper-controller.magma.wavelabs.in:7444
Apr 17 12:53:15 debian magmad[12496]: INFO:root:Retrying bootstrap in 30 seconds
Apr 17 12:53:17 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:27 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:37 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service subscriberdb (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service mobilityd (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service mme (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service magmad (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service enodebd (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service pipelined (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service state (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:Metrics upload error for service sessiond (chunk 0)! [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:45 debian magmad[12496]: WARNING:root:Collect mobilityd Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:45 debian magmad[12496]: WARNING:root:Collect mme Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:45 debian magmad[12496]: INFO:root:Checking for upgrade...
Apr 17 12:53:45 debian magmad[12496]: INFO:root:Service is currently on package version 1.4.0-1615912803-6bca6402, ignoring upgrade to 0.0.0-0 because it is either equal or behind.
Apr 17 12:53:45 debian magmad[12496]: WARNING:root:Collect pipelined Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:45 debian magmad[12496]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] DNS resolution failed for service: bootstrapper-controller.magma.wavelabs.in:7444
Apr 17 12:53:45 debian magmad[12496]: INFO:root:Retrying bootstrap in 30 seconds
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:GetOperationalStates Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:GetServiceInfo Error for pipelined! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:GetServiceInfo Error for mme! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:GetOperationalStates Error for mme! [StatusCode.UNAVAILABLE] failed to connect to all addresses
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:Checkin Error! Failed to report states. [StatusCode.UNAVAILABLE] Received http2 header with status: 503
Apr 17 12:53:48 debian magmad[12496]: INFO:root:StateReporting (Checkin) failure threshold met, remediating...
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:Error making TCP connection: Unknown error -2, [Errno -2] Name or service not known
Apr 17 12:53:48 debian magmad[12496]: ERROR:root:StateReporting failure likely not due to invalid cert

**Reply by pmishra-wavelabs** [reply]
[orc8r_logs.txt](https://github.com/magma/magma/files/6329365/orc8r_logs.txt)

**Reply by pmishra-wavelabs** [reply]
![image](https://user-images.githubusercontent.com/70897856/115115823-1e869c00-9fb4-11eb-929d-8e62bdd7909c.png)
